LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -98,10 +101,21 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
             // ABP device does not reset the Fcnt so in relax mode we should reset for 0 (LMIC based) or 1
             var isFrameCounterFromNewlyStartedDevice = await DetermineIfFramecounterIsFromNewlyStartedDeviceAsync(loRaDevice, payloadFcntAdjusted, frameCounterStrategy);
 
+            var concentratorDeduplicationResult = await this.concentratorDeduplication.CheckDuplicateAsync(request, loRaDevice);
+            if (concentratorDeduplicationResult is ConcentratorDeduplication.Result.Duplicate)
+            {
+                return new LoRaDeviceRequestProcessResult(loRaDevice, request, LoRaDeviceRequestFailedReason.DeduplicationDrop);
+            }
+            else if (concentratorDeduplicationResult is ConcentratorDeduplication.Result.SoftDuplicate)
+            {
+                // Request is allowed upstream but confirmation is skipped to avoid collisions on the air.
+                requiresConfirmation = false;
```

Comment:
check the places where we set this flag. You absolutely don't want to send a message down in this case, hence, there should be no code path resetting this. However, for that, I believe you need a separate flag to indicate, that you want this to be enforced - for example, if it is a MacCommand that you already processed I think there are 2 issues:

1. we will potentially contact the server for a framecounter down
2. we will reset the flag and send the answer to the device again

Reply:
Done and covered with integration tests, resolving this.